### ISSUE TITLE
Support SimpleNamespace

### ISSUE THREAD

**Issue by kbehlers** [davidhalter/jedi#1645]
Currently `SimpleNamespace` doesn't populate autocomplete suggestions for set attributes. I've looked through the issues and haven't seen `SimpleNamespace` mentioned. Could be a similar problem to `Enum`, `namedtuple`, or `dataclass`?

FWIW I see the [same issue](https://github.com/microsoft/python-language-server/issues/2108) when using the Microsoft Python Language Server in VSCode, but Jedi actually gives me different Intellisense options if you compare the screenshots between the two issues.

## Environment data
VSCode

> Version: 1.47.2
> Commit: 17299e413d5590b14ab0340ea477cdd86ff13daf
> Date: 2020-07-15T18:18:50.054Z
> Electron: 7.3.2
> Chrome: 78.0.3904.130
> Node.js: 12.8.1
> V8: 7.8.279.23-electron.0
> OS: Darwin x64 18.7.0

Python Extension

> Name: Python
> Id: ms-python.python
> Version: 2020.7.96456
> Publisher: Microsoft

Language Server Settings
```
{
  "python.languageServer": "Jedi",
}
```

## Code Example
```python
from types import SimpleNamespace
colors = SimpleNamespace(red='red', blue='blue')
# this should autocomplete, it doesn't find `red` or `blue` as options
colors.red
```

### Screenshot
![image](https://user-images.githubusercontent.com/43154039/88948743-898a1a00-d24f-11ea-8806-d4cc736e6f3a.png)

**Comment by davidhalter** [comment]
Didn't even know this existed :)

**Comment by davidhalter** [comment]
I'm not sure I'm going to implement this. This seems like an old concept that was "invented" before all of the typing stuff. It's probably better to use namedtuples or dataclasses for this.

**Comment by kbehlers** [comment]
@davidhalter Yeah I could understand that. It may very well be an old concept, but it fits very nicely in certain niche cases.

My use case currently is for grouping several related constants under a "simple namespace."
```
# constants.py
from types import SimpleNamespace
Stage = SimpleNamespace(
    REQUESTED='requested', IN_PROCESS='checked_out', SHIPPING='past'
)

# main.py
from .constants import Stage
if item.stage == Stage.REQUESTED:
    # do something
    return
```

I've also used it on occasion as a one-off object when I need to duck type another object.

[`venv` uses it] to build a configuration context object on the flyhttps://github.com/python/cpython/blob/58ec58a42bece5b2804b178c7a6a7e67328465db/Lib/venv/__init__.py#L111

Basically SimpleNamespace gives:
1. access by dot notation (unlike dict)
2. comparison by value by default rather than identity, so I don't have to have to pass a singleton among all my dependencies (unlike Enum)
3. direct instantiation rather than subclassing then instantiation (e.g. NamedTuple and dataclass)
4. kwarg init so SyntaxError is raised on duplicate init kwargs

Small things, possible in other types, but all together make for that extra bit of sugar.

**Comment by davidhalter** [comment]
I'm choosing not to implement this. I think it's a fine feature, but I just have other things that have priority.

So if you want to implement this, I'm happy to assist, it shouldn't be a very big change. If you don't want to implement it, I will probably close this ticket.

**Comment by kbehlers** [comment]
@davidhalter Thanks for considering it. I'll read through the [developer docs](https://jedi.readthedocs.io/en/latest/docs/development.html) this week and try my hand at implementing it. If you have any pointers to start let me know.

**Comment by kbehlers** [comment]
If someone comes along that wants this, https://github.com/davidhalter/jedi/pull/1649 started down the path.
